### PR TITLE
docs: update MVUX counter

### PR DESCRIPTION
**Pull request by rajamatt**
GitHub Issue (If applicable): #https://github.com/unoplatform/private/issues/446

see https://github.com/unoplatform/Uno.Samples/pull/693 for changes

## PR Type

What kind of change does this PR introduce?

- Documentation content changes

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: doc/articles/getting-started/counterapp/includes/include-mvux.md [hunk 1]
```diff
@@ -13,35 +13,62 @@ As part of creating the application, we selected MVUX as the presentation framew
...
+    };
+    ```
+
+- Add the newly created `Countable` as a property in the `MainModel` class. The type must be `IState<Countable>` and we use `=> State.Value(...)` to initialize it.
+
+    ```csharp
+    public IState<Countable> Countable => State.Value(this, () => new Countable(0, 1));
+    ```
+
+- Add a method named `IncrementCommand` to the `MainModel` that will in turn call the `Countable`'s `Increment` method and therefore update the counter. The generated ViewModel of our `MainModel` will automatically re-expose as `ICommand` the `IncrementCommand` method.
```
Comment (ajpinedam) [inline doc/articles/getting-started/counterapp/includes/include-mvux.md:47]: Is it important at this point to mention the Generated MainModel and the ICommand?

Comment (rajamatt) [reply doc/articles/getting-started/counterapp/includes/include-mvux.md:47]: good point, I think we could instead mention it here when we are databinding: https://github.com/unoplatform/uno/blob/7f009e1c3fa6e4acb07e92ece2cd4ef62524b0bf/doc/articles/getting-started/counterapp/get-started-counter-csharp-mvux.md?plain=1#L126

(and here) https://github.com/unoplatform/uno/blob/7f009e1c3fa6e4acb07e92ece2cd4ef62524b0bf/doc/articles/getting-started/counterapp/get-started-counter-xaml-mvux.md?plain=1#L122

Comment (rajamatt) [reply doc/articles/getting-started/counterapp/includes/include-mvux.md]: @ajpinedam I have removed the mention and added a reference to commands in MVUX instead

File: doc/articles/getting-started/counterapp/includes/include-mvux.md [hunk 2]
```diff
@@ -13,35 +13,62 @@ As part of creating the application, we selected MVUX as the presentation framew
     }
     ```
 
-- Add the `Count` and `Step` properties, to the `MainModel` class. These fields both must be of type `IState<int>` and to initialize them we use `=> State.Value(...)`.
+- Add a new `partial record` above named `Countable`. This record will be responsible for updating our counter's properties all while ensuring immutability. Learn more about immutable records [here](https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Mvux/WorkingWithRecords.html#how-to-create-immutable-records).
```
Comment (agneszitte) [inline doc/articles/getting-started/counterapp/includes/include-mvux.md:16]:
```suggestion
- Add a new `partial record` above named `Countable`. This record will be responsible for updating our counter's properties all while ensuring immutability. Learn more about immutable records [here](xref:Uno.Extensions.Mvux.Records#how-to-create-immutable-records).
```

Comment (agneszitte) [reply doc/articles/getting-started/counterapp/includes/include-mvux.md:16]: @rajamatt  Make sure to verify that the link is properly working while running the doc locally please

File: doc/articles/getting-started/counterapp/includes/include-mvux.md [hunk 3]
```diff
@@ -13,35 +13,62 @@ As part of creating the application, we selected MVUX as the presentation framew
...
+- Add the `Count` and `Step` properties to the `Countable`'s primary constructor.
+
+    ```csharp
+    internal partial record Countable(int Count, int Step)
+    {
+    }
     ```
 
-- Add a method called `IncrementCommand` to the `MainModel` that will increment the counter by the step size. The generated ViewModel of our `MainModel` will automatically re-expose as `ICommand` the `IncrementCommand` method.
+- Add an `Increment` method to the `Countable` record. The `with` operator allows us to [create a new instance](https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Mvux/WorkingWithRecords.html#updating-records) of the object.
```
Comment (agneszitte) [inline doc/articles/getting-started/counterapp/includes/include-mvux.md:32]:
```suggestion
- Add an `Increment` method to the `Countable` record. The `with` operator allows us to [create a new instance](xref:Uno.Extensions.Mvux.Records#updating-records) of the object.
```

Comment (agneszitte) [reply doc/articles/getting-started/counterapp/includes/include-mvux.md:32]: @rajamatt Make sure to verify that the link is properly working while running the doc locally please

File: doc/articles/getting-started/counterapp/includes/include-mvux.md [hunk 4]
```diff
@@ -13,35 +13,62 @@ As part of creating the application, we selected MVUX as the presentation framew
...
+    };
+    ```
+
+- Add the newly created `Countable` as a property in the `MainModel` class. The type must be `IState<Countable>` and we use `=> State.Value(...)` to initialize it.
+
+    ```csharp
+    public IState<Countable> Countable => State.Value(this, () => new Countable(0, 1));
+    ```
+
+- Add a method named `IncrementCounter` to the `MainModel` that will in turn call the `Countable`'s `Increment` method and therefore update the counter. You can find more information on commands in MVUX [here](https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Mvux/Advanced/Commands.html).
```
Comment (agneszitte) [inline doc/articles/getting-started/counterapp/includes/include-mvux.md:47]:
```suggestion
- Add a method named `IncrementCounter` to the `MainModel` that will in turn call the `Countable`'s `Increment` method and therefore update the counter. You can find more information on commands in MVUX [here](xref:Uno.Extensions.Mvux.Advanced.Commands).
```

**Review by dr1rrb — APPROVED** [review]

**Comment by agneszitte** [comment]
@Mergifyio backport release/stable/5.2

**Pull request merged**